src/interpreter.ts

```diff
@@ -0,0 +1,176 @@
+import { Path, PathType } from './path';
+import { Code, GCodeCommand } from './gcode-parser';
+import { Machine } from './machine';
+
+export class Interpreter {
```

Comment:
It would be cool for lib users to be able to use their own custom interpreter (or override this one). The [features around distances on the CNC features PR](https://github.com/remcoder/gcode-preview/pull/183/files#diff-8ec787c979ef6cb88e426e798a75cc41d326888eaca61c5f5691b18c5e65c729R174-R181) could be possible as a custom interpreter instead of an official part of this library. 

There could also be some slicer specific interpreter dictionaries implemented in the interpreter to be able to implement some preview features more easily. For example, comments to mark extrusion types (infill, outer wall, wipe) are different depending on the different slicers.

Reply:
> For example, comments to mark extrusion types (infill, outer wall, wipe) are different depending on the different slicers.

Actually, that should be in the parser and formatted in a standard way for the interpreter to interpret.

Reply:
Do you think we'll get away with a customizable parser but a generic interpreter? I think for the most part yes but I trying to come up with a counter example... can't think of one atm..

The parser could be a constructor param (injection) and if not present, we instantiate the default parser. This would be backwards compatible, so it could be a v2 feature.

Reply:
I think that anything related to calculating (the distance for example) should be in the interpreter. 

And yes, I'm thinking about injection/default, for both the parser and the interpreter. There's value into having custom in all cases, as long as the AST is solid